src/common/transformations/src/transformations/common_optimizations/gelu_fusion.cpp

```diff
@@ -333,3 +333,91 @@ ov::pass::GeluFusionWithTanhNoPower::GeluFusionWithTanhNoPower() {
+    // Gaussian Error Linear Unit, TanH based approximation:
+    // x * (0.5 * (1 + tanh([sqrt(2 / pi)] * [x + 0.044715 * x * x * x])))
+
+    auto input = pass::pattern::any_input();
+
+    auto mul_0 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, input});
+    auto mul_1 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, mul_0});
+
+    auto mul_2_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>();
+    auto mul_2 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({mul_1, mul_2_constant});
+
+    auto add_0 = ov::pass::pattern::wrap_type<ov::op::v1::Add>({input, mul_2});
+
+    auto mul_3_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>();
+    auto mul_3 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_0, mul_3_constant});
+
+    auto tanh = ov::pass::pattern::wrap_type<ov::op::v0::Tanh>({mul_3});
+
+    auto add_1_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>();
+    auto add_1 = ov::pass::pattern::wrap_type<ov::op::v1::Add>({tanh, add_1_constant});
+
+    auto mul_4_constant = ov::pass::pattern::wrap_type<ov::op::v0::Constant>();
+
+    // x * (0.5 * (1 + tanh))
+    auto mul_4_1 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_1, mul_4_constant});
+    auto mul_5_1 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, mul_4_1});
+
+    // (x * 0.5) * (1 + tanh)
+    auto mul_4_2 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({input, mul_4_constant});
+    auto mul_5_2 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_1, mul_4_2});
+
+    // (0.5 * x) * (1 + tanh)
+    auto mul_4_3 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({add_1, input});
+    auto mul_5_3 = ov::pass::pattern::wrap_type<ov::op::v1::Multiply>({mul_4_3, mul_4_constant});
+
+    auto mul_5 = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{mul_5_1, mul_5_2, mul_5_3});
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        auto& pattern_to_output = m.get_pattern_value_map();
+        auto x_output = pattern_to_output.at(input);
+
+        auto mul_2_constant_value =
+            ov::as_type_ptr<ov::op::v0::Constant>(pattern_to_output.at(mul_2_constant).get_node_shared_ptr());
+        auto mul_3_constant_value =
+            ov::as_type_ptr<ov::op::v0::Constant>(pattern_to_output.at(mul_3_constant).get_node_shared_ptr());
+        auto mul_4_constant_value =
+            ov::as_type_ptr<ov::op::v0::Constant>(pattern_to_output.at(mul_4_constant).get_node_shared_ptr());
+        auto add_1_constant_value =
+            ov::as_type_ptr<ov::op::v0::Constant>(pattern_to_output.at(add_1_constant).get_node_shared_ptr());
+        if (!add_1_constant_value || !mul_2_constant_value || !mul_3_constant_value || !mul_4_constant_value) {
+            return false;
+        }
+
+        bool valid_constant_values =
+            op::util::has_constant_value<float>(mul_2_constant_value, 0.044715f, 0.001f) &&
```

Comment:
Please move the constants validation to the pattern as in GeluFusionWithErfThree
<img width="386" alt="{9D001399-5982-42D2-AEF8-631F61ADEEAC}" src="https://github.com/user-attachments/assets/f4b6e5c0-e6c7-4e46-a582-f01879951f7e" />
it would simplify callback code and allows as to use commutative property of Mul nodes

Reply:
could you use the know math constants form math.h or cmath if it's possible?

Reply:
Applied, thanks!